neuralprophet/TorchProphet.py

```diff
@@ -0,0 +1,553 @@
+                periods=periods,
+                n_historic_predictions=include_history,
+                **kwargs,
+            )
+        else:
+            df_future = super(TorchProphet, self).make_future_dataframe(
+                df=self.history, periods=periods, n_historic_predictions=include_history, **kwargs
+            )
+        return df_future
+
+    def add_seasonality(self, name, period, fourier_order, prior_scale=None, mode=None, condition_name=None, **kwargs):
+        """Add a seasonal component with specified period, number of Fourier
+        components, and prior scale.
+
+        Increasing the number of Fourier components allows the seasonality to
+        change more quickly (at risk of overfitting). Default values for yearly
+        and weekly seasonalities are 10 and 3 respectively.
+
+        Increasing prior scale will allow this seasonality component more
+        flexibility, decreasing will dampen it. If not provided, will use the
+        seasonality_prior_scale provided on Prophet initialization (defaults
+        to 10).
+
+        Mode can be specified as either 'additive' or 'multiplicative'. If not
+        specified, self.seasonality_mode will be used (defaults to additive).
+        Additive means the seasonality will be added to the trend,
+        multiplicative means it will multiply the trend.
+
+        If condition_name is provided, the dataframe passed to `fit` and
+        `predict` should have a column with the specified condition_name
+        containing booleans which decides when to apply seasonality.
+
+        Parameters
+        ----------
+        name: string name of the seasonality component.
+        period: float number of days in one period.
+        fourier_order: int number of Fourier components to use.
+        prior_scale: Not supported in NeuralProphet.
+        mode: optional 'additive' or 'multiplicative'
+        condition_name: Not supported in NeuralProphet.
+
+        Returns
+        -------
+        The prophet object.
+        """
+        # Check for unsupported features
+        if condition_name:
+            raise NotImplementedError("Conditioning on seasonality is not supported in NeuralProphet.")
+        if prior_scale:
+            log.warn(
+                "Prior scale is not supported in NeuralProphet. Use the `regularisation` parameter for regularisation."
+            )
+        # Set attributes in NeuralProphet config
+        try:
+            self.season_config.mode = mode
+        except AttributeError:
+            log.warn("Cannot set the seasonality mode attribute in NeuralProphet. Pleas inspect manually.")
+        # Run the NeuralProphet function
+        return super(TorchProphet, self).add_seasonality(name, period, fourier_order, **kwargs)
+
+    def add_regressor(self, name, prior_scale=None, standardize="auto", mode="additive", **kwargs):
+        """Add an additional regressor to be used for fitting and predicting.
```

Comment:
Please adopt NP docstring as far as reasonable / remove descriptions of parameters not supported